DigitalLearningSolutions.Web/ViewModels/Common/SearchablePage/BasePaginatedViewModel.cs

```diff
@@ -1,42 +1,54 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage
 {
     using System;
-    using System.Collections;
     using System.Collections.Generic;
     using System.Linq;
     using DigitalLearningSolutions.Data.Models;
     using DigitalLearningSolutions.Web.Helpers;
+    using Microsoft.AspNetCore.Mvc.Rendering;
 
     public abstract class BasePaginatedViewModel
     {
         public const int DefaultItemsPerPage = 10;
-        private readonly int itemsPerPage;
 
         public int MatchingSearchResults;
 
         protected BasePaginatedViewModel(int page, int itemsPerPage = DefaultItemsPerPage)
         {
             Page = page;
-            this.itemsPerPage = itemsPerPage;
+            ItemsPerPage = itemsPerPage;
         }
 
         public int Page { get; protected set; }
 
         public int TotalPages { get; protected set; }
 
+        public int ItemsPerPage { get; protected set; }
+
+        public IEnumerable<SelectListItem> ItemsPerPageSelectListItems =>
+            SelectListHelper.MapOptionsToSelectListItems(ItemsPerPageOptions);
+
+        public virtual IEnumerable<(int, string)> ItemsPerPageOptions { get; } = new[]
+        {
+            (10, "10"),
+            (25, "25"),
+            (50, "50"),
+            (100, "100"),
+        };
+
         protected IEnumerable<T> GetItemsOnCurrentPage<T>(IList<T> items)
         {
-            if (items.Count > itemsPerPage)
+            if (items.Count > ItemsPerPage)
             {
-                items = items.Skip(OffsetFromPageNumber(Page)).Take(itemsPerPage).ToList();
+                items = items.Skip(OffsetFromPageNumber(Page)).Take(ItemsPerPage).ToList();
             }
 
             return items;
         }
 
         protected void SetTotalPages()

```

Comment:
I know this is how it was set up in the past, not by you, but it feels odd to me that we have this SetTotalPages method instead of having the definition of TotalPages automatically set it, and then maybe renaming this method and only having it be responsible for setting Page to 1 when it is greater than TotalPages.

Reply:
I know what you mean.
There are a 13 places where this method is used so a fair number of places that need re-testing.
This is sounding like a `Tech Debt` ticket is required. What do you think?

Reply:
Yeah, not worth making it part of this ticket in that case. I'll raise a tech debt ticket to cover refactoring this.